client/scripts/views/components/inline_thread_composer.ts

```diff
@@ -178,10 +186,45 @@ const TextPost: m.Component<ITextPostAttrs, ITextPostState> = {
             onclick: createThread,
             tabindex: 4
           }, 'Create thread'),
-          // m('button', {
-          //   type: 'cancel',
-          //   onclick: closeComposer,
-          // }, 'Cancel'),
+          m('.property-group', [
+            m('input[type="radio"]', {
+              name: 'properties',
+              value: 'public',
+              id: 'public-thread',
+              checked: (vnode.state.read_only === false && vnode.state.privacy === false),
+              onclick: () => {
+                vnode.state.read_only = false;
+                vnode.state.privacy = false;
+              }
+            }),
+            m('label', {
+              for: 'public-thread',
+            }, 'Public'),
+            m('input[type="radio"]', {
```

Comment:
We could definitely start consolidating these inputs into a separate component file, since they're using the same data except for a change in the logic of `onclick`. Thoughts are making resuable components?

Reply:
For example:
```
const createRadioButtonForThread = (name, value, id, vnode) => {
  let onclick: () => void;

  if (value === 'readOnly') {
    onclick = () => {
      vnode.state.read_only = true;
      vnode.state.privacy = false;
    }
  } else if (value === 'private' ) {
    onclick = () => {
      vnode.state.read_only = false;
      vnode.state.privacy = true;
    }
  }

  return m('input[type="radio"]', { name, value, id, onclick }),
}

createRadioButton('properties', 'readOnly', 'read-only')
createRadioButton('properties', 'private', 'private-thread')
```
thoughts @raykyri 

Similarly, our need to keep switching from snake case to camelCase is really not sensical. If we stuck to camelcases we could do things like `vnode.state[value] = true`.

Reply:
I think this is a pretty simple piece of logic and it's easier to read inline, rather than factored out, which increases the amount of complexity and logic that needs to be understood to read the code. In general I think it is best to treat UI components as configuration so a longer file would be okay (but we should be wary of complexity or gotchas like manipulations of `vnode.state`, DOM manipulation, or sidestepping Mithril).

I agree on the camelCase. We should be using camelCase on the frontend for attrs/state and this should noted by the linter too.

Reply:
Ok works for me, I think you're right.